### DISCUSSION TITLE
Wasm "Failed to load resource" and "WebSocket connection to 'wss://169.254.110.102:55118/rc' failed" issues

### DISCUSSION THREAD

**Discussion by adampulley** [unoplatform/uno#8539] [Q&A]
Hi, uno newb here converting my first UWP application to run in a browser using wasm. If I run my uno project in VS2019 as MyProject.UWP (Universal Windows) then it runs as expected.  However when I run it as MyProject.Wasm it doesn't get past the initial default "Hello World" screen in the browser, that I've not yet removed, and I get lots of browser console errors:

![Capture](https://user-images.githubusercontent.com/5228153/162949004-0f2e554a-05f6-4dff-aa38-ce4f9a78290f.PNG)

I expect I've missed some setup steps and hoping someone may be able to point me in the right direction to resolve?
Thanks
Adam

**Comment by jeromelaban** [comment]
Thanks for the report. The WebSockets exceptions are for now expected and are caused by the Uno.UI.RemoteControl package for XAML hot reload, same for the "blazor-hotreload.js" error.

For the other errors, those are some that we haven't removed yet, as they are coming from the default WinUI templates. Let's keep track with https://github.com/unoplatform/uno/issues/8540.

**Reply by adampulley** [reply]
Ah OK thank you, I'd assumed any errors would mean something was up. So I'm wondering what the best way for me would be to diagnose what the issue is with my app when running as wasm as it doesn't run past its initial default Hello World screen currently, but does work correctly when running as UWP (I see the next screen appear that has all my apps functionality on it as I'd expect).

**Reply by jeromelaban** [reply]
Do you mean that the app does not show the "Hello world!" text?

**Reply by adampulley** [reply]
So I've left the default Hello World text in, but my app then changes to a new page after carrying out all sorts of initialisation functionality. When I run as wasm that page change doesn't occur, indicating something has gone wrong (when I run as uwp it does).  My first thought was to look at the console log to see what errors appeared, but as errors are expected there I need another way to try and find out whats causing it to fail to move on to the new page loading. Being a uno novice I'm not sure what the best way to go about that would be

**Reply by jeromelaban** [reply]
No problem at all. If you're navigating to another page, it may be that you're getting an exception in during the navigation. If you can, are you able to post a zip file with your sample code here?

**Reply by adampulley** [reply]
Oh OK thanks, it's a large project and client sensitive so I can't distribute unfortunately.  I'll have a play around and see what I can figure out..thank you for your responses

**Reply by jeromelaban** [reply]
No problem. If you can't share, make sure to find all the locations where exceptions may be hidden, like with Frame.NavigationFailed, or other similar exception raising through events or method overrides.

**Reply by adampulley** [reply]
Thank you

**Reply by adampulley** [reply]
So as a bit of feedback to anyone else with similar issues, I solved by using Console.WriteLine() to output debug messages to the browsers console and then narrowed down to finding the bits of my code that were causing the issue under uno wasm
